components/tikv_alloc/src/lib.rs

```diff
@@ -77,24 +77,28 @@
 //! jemalloc is configured for profiling if you pass
 //! `--features=mem-profiling` to cargo for eather `tikv_alloc` or
 //! `tikv`.
+mod default;
 
 #[cfg(feature = "mem-profiling")]
 #[macro_use]
 extern crate log;
 
-#[cfg(not(all(unix, not(fuzzing), feature = "jemalloc")))]
-mod default;
+#[cfg(not(any(feature = "jemalloc", feature = "tcmalloc", feature = "system-alloc")))]
+compile_error!("Features must be enabled for this crate.");
```

Comment:
Could we make `system-alloc` an implicit default? That would let us avoid this and the need for --no-default-features

Reply:
@brson  PTAL

Reply:
I don't see how it would avoid the need for `--no-default-features`. If "system-alloc" is the implicit default, then tikv is going to set `default = ["tikv_alloc/jemalloc"]`, so to configure tikv to use a different allocator `--no-default-features` is still needed.

If _"jemalloc"_ was the implicit default, then tikv wouldn't need to select an allocator, and `--no-default-features` would not be necessary. Sadly, a good amount of the effort that has gone into adding multiple allocator support has also gone toward not making jemalloc privileged in this crate (ultimately I would like this crate to not be tikv-specific), so it would be a big reversal to set jemalloc as the implicit default again.

To get around having `--no-default-features` on every makefile cargo invocation we could also simply not make the allocator configurable via the makefile.

Reply:
I was thinking that we could set the cfgs in such a way that if there is specified allocator, we use the system allocator, and if one is specified, then we use it without any need for `--no-default-features` (tikv would still need `--no-default-features`, but tikv_allocator would not).